### ISSUE TITLE
why usePromise.promise.value is null?

### ISSUE THREAD

**Issue by wizardnet972** [pikax/vue-composable#194]
I try to get the promise because I want to attach hook to the `then` callback. but I get undefined. 

Is there a way to get the promise immediately after `usePromise` execute?

 ```
 function timeout(ms) {
    return new Promise((resolve) => setTimeout(resolve, ms));
  }

  const api = usePromise(() => Promise.all([timeout(12000).then(() => ({ some: 'data'}))]));

  console.log({ p: api.promise.value }); // <---- undefined
```

[code in codesandbox](https://codesandbox.io/s/romantic-goldwasser-88q1w)

**Comment by pikax** [comment]
Currently the `usePromise` is only executed after `exec()` is called.

The reason for that is `usePromise` allows you to `exec` with arguments
```ts
const {exec} = usePromise((page:number)=> api.fetchPage(page));

exec(1);
exec(2);
```

But I'm thinking in changing the behaviour to execute immediately and add a options argument to allow to do lazy. But on the example above the `exec` would be called with `undefined`

My thoughts currently are:

```ts
usePromise(()=>api()); // immediately 
usePromise(()=>api(), {immediately: false}); // lazy

usePromise((...args)=>api(...args)); // lazy - if contains args 
usePromise((...args)=>api(...args), {immediately: true}); // immediately 
```

My concern is the different behaviour of having `args` and not having them, but if I exec them immediately might cause some problems.

**Comment by wizardnet972** [comment]
@pikax

**Comment by pikax** [comment]
Released [1.0.0-dev.16](https://github.com/pikax/vue-composable/releases/tag/v1.0.0-dev.16)

Update the [documentation](https://pikax.me/vue-composable/composable/promise/promise.html) to reflect the changes

I decided to make the `usePromise` to exec straight away, if the `factory` has argument, a warning will be log. 

Also added a sugar for lazy promises [usePromiseLazy](https://pikax.me/vue-composable/composable/promise/promiseLazy)


Let me know if you encounter any other issue 👍

**Comment by wizardnet972** [comment]
@pikax thank you :)
